docs/HowToConnectExternally.md

```diff
@@ -0,0 +1,298 @@
+DOMAIN_ADDRESS=$(kubectl get services nuodb-balancer -o jsonpath='{.status.loadBalancer.ingress[].ip}')
+```
+
+Check database processes node IDs:
+
+```shell
+kubectl exec -ti admin-nuodb-cluster0-0 -- nuocmd show database \
+    --db-name demo \
+    --process-format '{type} {hostname} {node_id}'
+```
+
+Use `nuosql`, which can be found in the [NuoDB Client-only Package](https://github.com/nuodb/nuodb-client), to connect to the NuoDB database from the local machine.
+Repeat the command several times to ensure that each time the expected node ID is printed.
+
+```shell
+echo 'select GETNODEID() from dual;' |  nuosql demo@${DOMAIN_ADDRESS} \
+    --user dba --password secret \
+    --connection-property 'LBQuery=round_robin(first(label(tx-type transactional) any))'
+```
+
+Repeat the steps for the _reporting_ workload.
+
+```shell
+echo 'select GETNODEID() from dual;' |  nuosql demo@${DOMAIN_ADDRESS} \
+    --user dba --password secret \
+    --connection-property 'LBQuery=round_robin(first(label(tx-type reporting) any))'
+```
+
+Verify that internal SQL clients can still connect to the database by setting `PreferInternalAddress=true` connection property.
+
+```shell
+kubectl exec -ti admin-nuodb-cluster0-0 -- bash -c \
+    "echo 'select GETNODEID() from dual;' |  nuosql demo@nuodb-clusterip \
+        --user dba --password secret \
+        --connection-property 'LBQuery=round_robin(first(label(tx-type transactional) any))' \
+        --connection-property 'PreferInternalAddress=true'"
+```
+
+## Cloud Provider Specifics
+
+### Native CNI
+
+All of the managed Kubernetes offerings from cloud providers supported by the NuoDB Helm Charts (EKS, AKS, GKE) have the ability to deploy Kubernetes with a native Container Network Interface (CNI) plugin.
+This causes Kubernetes pods to be assigned IP addresses from the underlying virtual network CIDR, making them addressable by SQL clients running in the same VPC (or peered VPCs / VPN connections with the correct routing configured).
+As such, the TE IP address returned by the second stage of the client connection protocol is already addressable by the client, and it is enough to make only the Admin layer available external to the Kubernetes cluster using the `admin.externalAccess.*` values, to receive client connections.  
+
+> **NOTE**: The node security configuration may also need to explicitly allow the NuoDB TE port (48006).
+
+### GCP
+
+No additional configuration is needed to enable NuoDB database external access in GKE.
+For more information, check [Configuring TCP/UDP load balancing](https://cloud.google.com/kubernetes-engine/docs/how-to/service-parameters).
+
+### AWS
+
+By default, the NuoDB Helm Charts 3.4.0 excepts that the [AWS Load Balancer Controller](https://docs.aws.amazon.com/eks/latest/userguide/aws-load-balancer-controller.html) is deployed in the Amazon Elastic Kubernetes Service (EKS) and the native CNI plugin is used when provisioning internet-facing load balancer.
```

Comment:
Should say "expects" instead of "excepts".

Reply:
Done